### PR TITLE
Deprecated napalm_mod.py, netmiko_conn and pyeapi_conn functions not to be called from the CLI

### LINKED ISSUE #61566: [TECH DEBT] netmiko_conn and pyeapi_conn need to be deprecated
**Description**
The netmiko_coon and pyeapi_conn are functions in salt/modules/napalm_mod.py which should be internal since there is no CLI usage, has a warning in about CLI usage in the doc string for the function.

**Setup**
The GitHub action doc bot caught this issue in a recent PR submission, noting that this function definition is 4 years old.  All public functions should be able to be used from the CLI.

- [ ] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD

  
**Expected behavior**
All public functions in modules should be able to be used from the CLI.

**Screenshots**
Capture of Slack conversation between @dmurphy18 and @s0undt3ch 
```
David Murphy:house_with_garden:  11:29 AM
I am talking about fixing the bot, not disable it, here is the doc string in the code
 694 @proxy_napalm_wrap
 695 def netmiko_conn(**kwargs):
 696     """
 697     .. versionadded:: 2019.2.0
 698 
 699     Return the connection object with the network device, over Netmiko, passing
 700     the authentication details from the existing NAPALM connection.
 701 
 702     .. warning::
 703 
 704         This function is not suitable for CLI usage, more rather to be used
 705         in various Salt modules.
 706 
 707     USAGE Example:
 708 
 709     .. code-block:: python
 710 
 711         conn = __salt__['napalm.netmiko_conn']()
 712         res = conn.send_command('show interfaces')
 713         conn.disconnect()
 714     """
That 'warning::' should disable the bot from expecting 'CLI Example', that is what needs to be fixed

Pedro Algarvio  11:33 AM
No, if it's a public function, it should have a CLI Example, if it's not a public function, it should be prefixed with _
11:33
by public I mean, something we run from the CLI
11:34
ie, that function is exposed on the loader, when it should not

David Murphy:house_with_garden:  11:38 AM
see what you are saying, but this is old code.
Rats, mircea 4 years old code, wonder who'd we'd break if alter that now.
Guess leave the bot alone, and ignore the warning, we'd need to deprecate the API, can next big release would be best time to do it.
Wonder how much other stuff like this is in the code and have not caught till now
New

Pedro Algarvio  11:39 AM
OR, make that a private function, update code that calls it, and add another function with the old name, which triggers a deprecation warning for the next 2 versions :grin:
11:39
the new functions redirects to the private function, and we ignore the warning

David Murphy:house_with_garden:  11:40 AM
Sounds like a plan

Pedro Algarvio  11:40 AM
AND, add the, now deprecated function to https://github.com/saltstack/salt/blob/master/tasks/docstrings.py#L22
GitHubGitHub
salt/docstrings.py at master · saltstack/salt
Software to automate the management and configuration of any infrastructure or application at scale. Get access to the Salt software package repository here: - salt/docstrings.py at master · salts... (58 kB)
https://github.com/saltstack/salt/blob/master/tasks/docstrings.py#L22

David Murphy:house_with_garden:  11:41 AM
will create an issue and self assign, get this done
```

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
PASTE HERE
```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Deprecated netmiko_conn and pyeapi_conn in napalm_mod.py as these function should not be called from the CLI after conversation with @s0undt3ch whereby all public functions should be callable from the CLI.  Moves these functions to internal with an external function named similarly, and marked as deprecated to be removed in the Chlorine release.  Want to get this into Phosphorous so removed by Chlorine timeframe

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/61566

### Previous Behavior
Interfaces publicly available

### New Behavior
Interfaces publicly available, but marked deprecated and to be removed by Chlorine

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by dmurphy18** [comment]
re-run all

**Review by s0undt3ch — DISMISSED** [review]

**Comment by dmurphy18** [comment]
re-run amazon
re-run windows

File: salt/modules/napalm_mod.py [hunk 1]
```diff
@@ -712,9 +763,13 @@ def netmiko_conn(**kwargs):
         res = conn.send_command('show interfaces')
         conn.disconnect()
     """
-    netmiko_kwargs = netmiko_args()
-    kwargs.update(netmiko_kwargs)
-    return __salt__["netmiko.get_connection"](**kwargs)
+    salt.utils.versions.warn_until(
+        "Chlorine",
+        "This 'napalm_mod.netmiko_conn' function as been deprecated and "
+        "its functionality removed, as such, it has been made an internal "
```
Comment (Ch3LL) [inline salt/modules/napalm_mod.py:769]:
```suggestion
        "will be removed in the {version} release, as such, it has been made an internal "
```

File: salt/modules/napalm_mod.py [hunk 2]
```diff
@@ -1105,8 +1160,13 @@ def pyeapi_conn(**kwargs):
         res1 = conn.run_commands('show version')
         res2 = conn.get_config(as_string=True)
     """
-    pyeapi_kwargs = pyeapi_nxos_api_args(**kwargs)
-    return __salt__["pyeapi.get_connection"](**pyeapi_kwargs)
+    salt.utils.versions.warn_until(
+        "Chlorine",
+        "This 'napalm_mod.pyeapi_conn' function as been deprecated and "
+        "its functionality removed, as such, it has been made an internal "
```
Comment (Ch3LL) [inline salt/modules/napalm_mod.py:1166]:
```suggestion
        "will be removed in the {version} release, as such, it has been made an internal "
```

File: changelog/61566.deprecated [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Deprecated netmiko_conn and pyeapi_conn in napalm_mod.py as these function should not be called from the CLI
+
```
Comment (Ch3LL) [inline changelog/61566.deprecated:2]: remove extra line

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/napalm_mod.py (hunk 2 above)
Comment (dmurphy18) [reply salt/modules/napalm_mod.py:1166]: So much for C & P from other uses of salt.utils.versions.warn_until :+1:

File: changelog/61566.deprecated (hunk 3 above)
Comment (dmurphy18) [reply changelog/61566.deprecated:2]: done

File: salt/modules/napalm_mod.py (hunk 2 above)
Comment (dmurphy18) [reply salt/modules/napalm_mod.py:1166]: Updated other usage of it  too

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**